1.max_projection/check_and_max_proj.sh

```diff
@@ -1,20 +1,35 @@
 #!/bin/bash
 
+git_root=$(git rev-parse --show-toplevel)
+if [ -z "$git_root" ]; then
+    echo "Error: Could not find the git root directory."
+    exit 1
+fi
```

Comment:
This looks like it might be handy in more than just this script. Consider elevating to a `poethepoet`, `Makefile`, or `justfile` configuration for reuse.